### ISSUE TITLE
FiskalyHttpClient#request() throws JsonSyntaxException sometimes while parsing response.

### ISSUE THREAD

**Issue by dmitry-ryadnenko-izettle** [fiskaly/fiskaly-sdk-java#7]
@prempador knows about this. Adding this issue here just to keep track on when it's resolved and so our workaround can be removed from the code.

When calling `FiskalyHttpClient#request()` and sending a malformed requests with bad fields (for example bad `amounts_per_vat_rate.vat_rate` or `amounts_per_vat_rate.amount` or `amounts_per_payment_type.amount`) then `FiskalyHttpClient#request()` throws `JsonSyntaxException` while parsing the response.

From a Slack conversation with @prempador.

I’m working with transaction API right now. So, today I made an “update transaction” request. Apparently it was a little bit malformed and I got an error back. Which is fine. But I got the error in a form of an exception when calling FiskalyHttpClient#request() and the exception is this: com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Invalid escape sequence at line 1 column 145 path $.message . I suppose FiskalyHttpClient#request()
 is not supposed to throw exceptions like this (it’s signature says that it throws either IOException, FiskalyHttpException or FiskalyClientException) and JsonSyntaxException is none of those. So I look into the raw response and the response is like this:
```
{
  "jsonrpc": "2.0",
  "error": {
    "code": -20000,
    "message": "HTTP error",
    "data": {
      "response": {
        "status": 400,
        "headers": {
          "X-Request-Id": [
            "0123456789"
          ]
        },
        "body": "eyJzdGF0dXNfY29kZSI6IDQwMCwgImVycm9yIjogIkJhZCBSZXF1ZXN0IiwgImNvZGUiOiAiRV9UWF9VUFNFUlQiLCAibWVzc2FnZSI6ICJhbW91bnRzX3Blcl92YXRfcmF0ZS4wLnZhdF9yYXRlOiBhbW91bnRzX3Blcl92YXRfcmF0ZS4wLnZhdF9yYXRlIG11c3QgYmUgb25lIG9mIHRoZSBmb2xsb3dpbmc6ICIxOSIsICI3IiwgIjEwLjciLCAiNS41IiwgIjAiCmFtb3VudHNfcGVyX3ZhdF9yYXRlLjAuYW1vdW50OiBEb2VzIG5vdCBtYXRjaCBwYXR0ZXJuICdeLT9cZCsoXC5cZHsyLDY0fSkkJwphbW91bnRzX3Blcl9wYXltZW50X3R5cGUuMC5hbW91bnQ6IERvZXMgbm90IG1hdGNoIHBhdHRlcm4gJ14tP1xkKyhcLlxkezIsNjR9KSQnCiJ9"
      }
    }
  },
  "id": "4c631350-c6eb-43ad-8db7-2251141e52c4"
}
```
And the body decoded from base64 is like this:
```
{"status_code": 400, "error": "Bad Request", "code": "E_TX_UPSERT", "message": "amounts_per_vat_rate.0.vat_rate: amounts_per_vat_rate.0.vat_rate must be one of the following: "19", "7", "10.7", "5.5", "0"
  amounts_per_vat_rate.0.amount: Does not match pattern '^-?\d+(\.\d{2,64})$'
  amounts_per_payment_type.0.amount: Does not match pattern '^-?\d+(\.\d{2,64})$'
  "}
```
Which is an invalid json it seems. So I suppose the problem is that the backend sends an invalid json there encoded in Base64, client tries to parse it, fails, and throws an unexpected exception. It’s not a big deal for us right now since this shouldn’t happen with good requests, but maybe something you want to look into, so it’s easier for everybody to troubleshoot problems like this in the future.

**Comment by prempador** [comment]
This issue should be fixed with v1.1.600.

**Comment by pgaubatz** [comment]
Hi @dmitry-ryadnenko-izettle this should be fixed with 0fae8cd9428663c75f0528945b8c2a71c9ee5143
